checkers/utils.go

```diff
@@ -307,3 +309,7 @@ func identOf(x ast.Node) *ast.Ident {
 		return nil
 	}
 }
+
+func NodeToString(set *token.FileSet, node ast.Node) string {
```

Comment:
if the project already has similar function plz, point my nose

Reply:
I'm not 100% sure that we need this func yet 👀

Reply:
so, what should we do?

Reply:
AFAIK, the receiver name node is either `*ast.Ident` or `*ast.StarExpr`. So instead of doing an AST->string->trimAsterisk you can do:
```go
func (c *typeDefFirstChecker) receiverType(e ast.Expr) string {
  switch e := e.(type) {
  case *ast.Pointer:
    return c.receiverType(e.X)
  case *ast.Ident:
    return e.Value
  default:
    panic("unreachable")
  }
}
```
The `trimAsterisk` function, if we even wanted it, should be a method of the checker type. All helpers that are not generally useful should be encapsulated as a checker detail. As a general rule: no top-level functions should be defined inside the checker file.

Reply:
Wow, I am not familiar with all type relations, maybe some map or guide exists? That makes `trimAsterisk` 
unnecessary

Reply:
I think the closest thing is Go spec.
Although it's not always the most precise source as it defines valid receiver in its grammar as a normal param, but you can't have a slice type for a receiver, for example.